ocaml/xapi/xapi_sr.ml

```diff
@@ -128,6 +130,28 @@ let scanning_thread () = Debug.with_thread_named "scanning_thread" (fun () ->
 		done)
 	) ()
 
+let update_physical_utilisation ~__context =
+	let srs = get_all_plugged_srs ~__context in
+	(* Filter the SRs which have SR_STATS capability *)
+	let rrd_srs = List.filter (fun sr ->
+		let sr_record = Db.SR.get_record_internal ~__context ~self:sr in
+		Smint.(has_capability Sr_stats (Xapi_sr_operations.features_of_sr ~__context sr_record))
+		) srs
+	in
+	(* Update the physical utilisation db field of SRs *)
+	List.iter (fun sr ->
+		let new_value = Rrdd.query_sr_ds ~sr_uuid:(Db.SR.get_uuid ~__context ~self:sr) ~ds_name:"physical_utilisation" in
```

Comment:
The design calls for some form of caching to save us from doing this work every time: http://xapi-project.github.io/rrdd/futures/sr-level-rrds.html